L1Trigger/TrackerTFP/plugins/ProducerKFin.cc

```diff
@@ -0,0 +1,207 @@
+  };
+
+  ProducerKFin::ProducerKFin(const ParameterSet& iConfig) :
+    iConfig_(iConfig)
+  {
+    const string& labelTTTracks = iConfig.getParameter<string>("LabelSFout");
+    const string& labelStubs = iConfig.getParameter<string>("LabelSF");
+    const string& branchAcceptedStubs = iConfig.getParameter<string>("BranchAcceptedStubs");
+    const string& branchAcceptedTracks = iConfig.getParameter<string>("BranchAcceptedTracks");
+    const string& branchLostStubs = iConfig.getParameter<string>("BranchLostStubs");
+    const string& branchLostTracks = iConfig.getParameter<string>("BranchLostTracks");
+    // book in- and output ED products
+    edGetTokenTTTracks_ = consumes<vector<TTTrack<Ref_Phase2TrackerDigi_>>>(InputTag(labelTTTracks, branchAcceptedTracks));
+    edGetTokenStubs_ = consumes<TTDTC::Streams>(InputTag(labelStubs, branchAcceptedStubs));
+    edPutTokenAcceptedStubs_ = produces<TTDTC::Streams>(branchAcceptedStubs);
+    edPutTokenAcceptedTracks_ = produces<StreamsTrack>(branchAcceptedTracks);
+    edPutTokenLostStubs_ = produces<TTDTC::Streams>(branchLostStubs);
+    edPutTokenLostTracks_ = produces<StreamsTrack>(branchLostTracks);
+    // book ES products
+    esGetTokenSetup_ = esConsumes<Setup, SetupRcd, Transition::BeginRun>();
+    esGetTokenDataFormats_ = esConsumes<DataFormats, DataFormatsRcd, Transition::BeginRun>();
+    esGetTokenLayerEncoding_ = esConsumes<LayerEncoding, LayerEncodingRcd, Transition::BeginRun>();
+    // initial ES products
+    setup_ = nullptr;
+    dataFormats_ = nullptr;
+    layerEncoding_ = nullptr;
+    //
+    enableTruncation_ = iConfig.getParameter<bool>("EnableTruncation");
+  }
+
+  void ProducerKFin::beginRun(const Run& iRun, const EventSetup& iSetup) {
+    // helper class to store configurations
+    setup_ = &iSetup.getData(esGetTokenSetup_);
+    if (!setup_->configurationSupported())
+      return;
+    // check process history if desired
+    if (iConfig_.getParameter<bool>("CheckHistory"))
+      setup_->checkHistory(iRun.processHistory());
+    // helper class to extract structured data from TTDTC::Frames
+    dataFormats_ = &iSetup.getData(esGetTokenDataFormats_);
+    // helper class to encode layer
+    layerEncoding_ = &iSetup.getData(esGetTokenLayerEncoding_);
+  }
+
+  void ProducerKFin::produce(Event& iEvent, const EventSetup& iSetup) {
+    const DataFormat& dfCot = dataFormats_->format(Variable::cot, Process::sf);
+    const DataFormat& dfZT = dataFormats_->format(Variable::zT, Process::sf);
+    // empty KFin products
+    TTDTC::Streams streamAcceptedStubs(dataFormats_->numStreams(Process::kf) * setup_->numLayers());
+    StreamsTrack streamAcceptedTracks(dataFormats_->numStreams(Process::kf));
+    TTDTC::Streams streamLostStubs(dataFormats_->numStreams(Process::kf) * setup_->numLayers());
+    StreamsTrack streamLostTracks(dataFormats_->numStreams(Process::kf));
+    // read in SFout Product and produce KFin product
+    if (setup_->configurationSupported()) {
```

Comment:
In not configurationSupported, a fatal error should be thrown, perhaps from Setup.cc.

Reply:
no that would be prevent us to put the code into official cmssw release since they run automated tests including unsupported configuration. Throwing fatal error leads to not passing the test. They declined to improve their test system since it would increase its complexity.